frontend/src/components/FieldOfLawMain.vue

```diff
@@ -30,7 +30,7 @@ watch(localModelValue, () => {
   emit("update:modelValue", localModelValue.value)
 })
 
```

Comment:
[Non-blocking]
I actually thought of removing all the logic of adding/removing nodes and also the "selectedNode" from the FieldOfLawMain component and just  have the modelValue here as single source of truth. When a search result is clicked in FieldOfLawSearch, the component emits an event and the modelValue ist updated. When a tree node is clicked in FieldOfLawTree, the same happens. And when a result in FieldOfLawDirectInputSearch was clicked, again the same. So the Main component would hold the modelValue, and all three components update this modelValue. We wouldn't need all these functions (setSelectedNode, removeSelectedNode, addFieldOfLaw, removeFieldOfLaw)in the main component for example and for each component just one emit (@update:modelValue) (instead all those other listeners). 

But as this a big refactoring and also out of scope of this Mangel ticket, this is just a nice to have comment. Also, I possible oversee things, which make a refactoring like this not possible.

Reply:
@leonie-koch if it makes sense, we can switch to v-model. However, it is still does not change the matter that we currently based on the design we always need: 
selected node -> The node that is currently shown on the tree
saved nodes (modelValue for instance) -> the field of law nodes that is saved in our backend. 
Based on the design we need to have both always, to once have a direct tree to the node, second: show the user if this node is already saved in document unit

Reply:
💡